### PR TITLE
HHH-14080 Informix union all support

### PR DESCRIPTION
**Pull request by VladoKuruc**
All *UnionSubclassTest tests pass locally

### CODE REVIEW

**Comment by gavinking** [comment]
This looks fine to me though I can't test it because I don't have Informix. The change would need to be replicated to H6.

File: hibernate-core/src/main/java/org/hibernate/dialect/InformixDialect.java [hunk 1]
```diff
@@ -312,4 +312,19 @@ public IdentityColumnSupport getIdentityColumnSupport() {
...
 		return bool ? "'t'" : "'f'";
 	}
+	@Override
+	public boolean supportsUnionAll() {
+		return true;
+	}
+
+	@Override
+	public String getSelectClauseNullString(int sqlType) {
+		String typeName = getTypeName( sqlType, 1, 1, 0 );
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/dialect/InformixDialect.java:322]: Would it be better to use `getCastTypeName()` here?

Comment (VladoKuruc) [reply hibernate-core/src/main/java/org/hibernate/dialect/InformixDialect.java]: getCastTypeName() has the same result after trim off. The original implementation is the same as in PostgreSQL81Dialect.

**Comment by VladoKuruc** [comment]
I closed this pull request because I created a new one #8027 based on the current version in the main branch.

**Pull request closed without merging**